app/src/main/java/appricottsoftware/clarity/fragments/BrowseFragment.java

```diff
@@ -25,5 +58,81 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     @Override
     public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         // Initialize view lookups, listeners
+        podcasts = new ArrayList<>();
+        channels = new ArrayList<>();
+
+        // Parse browse.json from assets to populate Channel array.
+        parseJSON();
+        populateBrowseGrid();
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        if(context instanceof PlayerInterface) {
+            playerInterface = (PlayerInterface) context;
+        } else {
+            Log.e(TAG, context.toString() + " must implement PlayerInterface");
+            throw new ClassCastException(context.toString() + " must implement PlayerInterface");
+        }
+    }
+
+    // Parses json file and adds contexts to Channel ArrayList
+    void parseJSON() {
+        try {
+            JSONObject obj = new JSONObject(loadJSONFromAsset());
+            JSONArray resp = obj.getJSONArray("results");
+            for (int i = 0; i < resp.length(); i++) {
+                Channel c = getGson().fromJson(String.valueOf(resp.getJSONObject(i)), Channel.class);
+                c.setUid(ClarityApp.getSession(getActivity()).getUserID());
```

Comment:
Can we make this a for each loop?

Reply:
I don't believe we can. I think the index i is necessary to getting the right JSONobject from the JSONArray variable.

Reply:
Sorry to hijack this thread, but there's an example of doing it in ChannelFragment and PlayerService. You create a new type (ArrayList<Channel>) and have the gson library deserialize into that type. Something along the lines of:
`TypeToken<ArrayList<Channel>> token = new TypeToken<ArrayList<Channel>>() {};
                    ArrayList<Channel> channels = ClarityApp.getGson().fromJson(response.getObject("results"), token.getType());`

Reply:
Damn it looks like JSONArray uses raw iterators. haha

Reply:
@ttony678 Might be best to follow @kkong006 's templates.

Reply:
That assumes getting a JSONArray from the parameter. I was under the assumption we'd only be sending back JSONobjects?

Reply:
💯

Reply:
Using @kkong006 's template. Leaving older code commented because it allows setting UID within the loop which might or might not be useful at this moment.